extensions/auth/openldap/openldap.go

```diff
@@ -0,0 +1,117 @@
+package openldap
+
+import (
+	"fmt"
+
+	management "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/rancher/shepherd/pkg/config"
+	"github.com/rancher/shepherd/pkg/session"
+)
+
+type OLDAPOperations interface {
+	Enable() error
+	Disable() error
+}
+
+const (
+	resourceType = "openldap"
+	schemaType   = "openLdapConfigs"
+)
+
+type OLDAP struct {
+	Config *Config
+
+	session *session.Session
+	client  *rancher.Client
+}
+
+// NewOLDAP constructs OLDAP struct after it reads Open LDAP from the configuration file
+func NewOLDAP(client *rancher.Client, ts *session.Session) (*OLDAP, error) {
+	ldapConfig := new(Config)
+	config.LoadConfig(ConfigurationFileKey, ldapConfig)
+
+	return &OLDAP{
+		Config:  ldapConfig,
+		session: ts,
+		client:  client,
+	}, nil
+}
+
+// Enable is a method function of OLDAP, makes a request to the action with the given
+// configuration values
+func (o *OLDAP) Enable() error {
+	var jsonResp map[string]interface{}
+
+	url := o.newActionURL("testAndApply")
+	enableActionInput, err := o.newEnableInputFromConfig()
+	if err != nil {
+		return err
+	}
+
+	err = o.client.Management.Ops.DoModify("POST", url, enableActionInput, &jsonResp)
+	if err != nil {
+		return err
+	}
+
+	o.session.RegisterCleanupFunc(func() error {
+		return o.Disable()
+	})
+
+	return nil
+}
+
+// Disable is a method function of OLDAP, makes a request to disable Open LDAP
+func (o *OLDAP) Disable() error {
+	var jsonResp map[string]any
```

Comment:
curious, it was previously declared as an interface but we declare it as `any` here.. should it be interface too or it doesnt matter?

Reply:
IIRC after Golang 1.18 or 1.19, any is the same thing with interface{}